### PR TITLE
[ Ritesh ] | BAH-3191 - Fix issue of publishing event after openmrs b…

### PR DESCRIPTION
**Pull request by riteshghiya90**
BAH-3191 : Fixed issues which gets created due to change in openmrs version in base image.
With new openmrs version, we can't create and advice with @Bean annotation. hence replacing it with <advice> annotation in advice.

### CODE REVIEW

**Comment by gsluthra** [comment]
Minor comment from my side. Rest seems fine.

File: api/src/test/java/org/bahmni/module/events/api/listener/PatientAdviceTest.java [hunk 1]
```diff
@@ -31,80 +34,99 @@
...
 
         Event event = eventArgumentCaptor.getValue();
         assertEquals(BahmniEventType.BAHMNI_PATIENT_CREATED, event.eventType);
     }
 
     @Test
-    public void shouldPublishUpdateEventGivenPatientIsUpdated() {
+    public void shouldPublishUpdateEventGivenPatientIsUpdated() throws NoSuchMethodException {
         Patient newPatient = getPatient();
+        Method savePatientMethod = PatientService.class.getMethod("savePatient", Patient.class);
```
[the remark this answers is not in the document]
Comment (riteshghiya90) [reply api/src/test/java/org/bahmni/module/events/api/listener/PatientAdviceTest.java:100]: I have done a commit to fix that. Kindly review. I have created a new Set of methods name on which we have to apply the advice

[the remark this answers is not in the document]
Comment (gsluthra) [reply api/src/test/java/org/bahmni/module/events/api/listener/PatientAdviceTest.java:100]: I still see code which says `"savePatient"` here in tests. This won't refer to a constant?

[the remark this answers is not in the document]
Comment (riteshghiya90) [reply api/src/test/java/org/bahmni/module/events/api/listener/PatientAdviceTest.java]: Sorry i missed the suffix test from classname. 
I looked at the implementation class and there i have made changes. 
Just now fixed in the test as well.
Thanks Gurpreet :)

File: api/src/main/java/org/bahmni/module/events/api/configuration/JMSEventPublisherConfiguration.java [hunk 2]
```diff
@@ -14,9 +12,9 @@
 
 import javax.jms.ConnectionFactory;
 
-@Conditional(EventPublishingToggleCondition.class)
+@Conditional(JMSEventPublishingToggleCondition.class)
```
Comment (angshu) [inline api/src/main/java/org/bahmni/module/events/api/configuration/JMSEventPublisherConfiguration.java:15]: could have been done without a class using

> @ConditionalOnExpression

Comment (riteshghiya90) [reply api/src/main/java/org/bahmni/module/events/api/configuration/JMSEventPublisherConfiguration.java:15]: Agree angshu but looks like @ConditionalOnExpression annotation is not present

Comment (riteshghiya90) [reply api/src/main/java/org/bahmni/module/events/api/configuration/JMSEventPublisherConfiguration.java:15]: It comes with spring-boot-autoconfigure library which is not present

**Review by gsluthra — APPROVED** [review]

**Pull request merged**